ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m

```diff
@@ -134,148 +121,25 @@ + (ADRegistrationInformation *)getRegistrationInformation:(id<MSIDRequestContext
     return info;
 }
 
-
-+ (void)copyCertificate:(SecCertificateRef __nullable * __nonnull)certificate
-               identity:(SecIdentityRef __nullable * __nonnull)identity
-                 issuer:(NSString * __nullable * __nonnull)issuer
-                context:(id<MSIDRequestContext>)context
-                error:(ADAuthenticationError * __nullable __autoreleasing * __nullable)error
++ (CFDictionaryRef)getWPJIdentityDict:(id<MSIDRequestContext>)context
+               certificateAuthorities:(NSArray<NSData *> *)authorities
+                                error:(NSError **)error
 {
-    OSStatus status = noErr;
-    ADAuthenticationError *adError = nil;
-    NSData *issuerData = nil;
-    NSDictionary *identityQuery = nil;
-    CFDictionaryRef result = NULL;
-    
-    *identity = nil;
-    *certificate = nil;
-    if (error)
-    {
-        *error = nil;
-    }
-    
-    *certificate = [self copyWPJCertificateRef:context error:&adError];
-    
-    if (adError)
-    {
-        if (error)
-        {
-            *error = adError;
-        }
-        
-        MSID_LOG_ERROR(context, @"Failed to retrieve WPJ client certificate from keychain. Error code: %ld", (long)adError.code);
-        goto _error;
-    }
-    
-    // If there's no certificate in the keychain, adError won't be set since this isn't an error condition.
-    if (!*certificate)
-    {
-        return;
-    }
-    
-    // In OS X the shared access group cannot be set, so the search needs to be more
-    // specific. The code below searches the identity by passing the WPJ cert as reference.
-    identityQuery = @{ (__bridge id)kSecClass : (__bridge id)kSecClassIdentity,
-                       (__bridge id)kSecReturnRef : (__bridge id)kCFBooleanTrue,
-                       (__bridge id)kSecReturnAttributes : (__bridge id)kCFBooleanTrue,
-                       (__bridge id)kSecAttrKeyClass : (__bridge id)kSecAttrKeyClassPrivate,
-                       (__bridge id)kSecValueRef : (__bridge id)*certificate
-                       };
-    
-    status = SecItemCopyMatching((__bridge CFDictionaryRef)identityQuery, (CFTypeRef*)&result);
+    NSDictionary *query = @{ (__bridge id)kSecClass : (__bridge id)kSecClassIdentity,
```

Comment:
Does it also return the private key?

Reply:
No it does not. You have to extract the certificate and private key from identity. Previous code created a separate query to get the private key which is not required as you have the identity retrieved already.